lib/bindings/python/rust/lib.rs

```diff
@@ -330,12 +330,14 @@ impl EtcdKvCache {
         py: Python<'p>,
         key: String,
         value: Vec<u8>,
-        lease_id: Option<i64>,
+        // Unused. The client's primary lease is always attached to the entry.
+        // We can't rename or remove this because it might break Python.
+        #[allow(unused)] lease_id: Option<i64>,
```

Comment:
I'd prefer to remove these, but due to lack of code coverage no python side, I'll assume we keep this as is for this PR. They appear largely unused (passing `None`) in the majority of call sites, except the recent CustomLease stuff added, at a glance.

---

If we're keeping this in to avoid breaking python things loudly, can we at least log a loud error or something here if a non-null lease_id is passed to these functions (and not used) ?

I think it will be harder to debug silent failures expecting that these will do something different when passed a lease_id from python side (but actually don't) - rather than debugging a loud error.

Reply:
Alternatively we can remove them here, assume we may break python things loudly, and immediately swarm to address it. I'm more afraid of time spent to discover and debug subtle errors.

Reply:
Good point. Removed. Will merge once 0.2.0 is tagged so we have time to find any Python calls I missed.

Reply:
Some possible missed spots on python side for future reference:

- https://github.com/search?q=repo%3Aai-dynamo%2Fdynamo+serve_endpoint+language%3APython&type=code
- https://github.com/search?q=repo%3Aai-dynamo%2Fdynamo+kv_create_or_validate+language%3APython&type=code
- https://github.com/search?q=repo%3Aai-dynamo%2Fdynamo+kv_put+language%3APython&type=code&l=Python

CC @ishandhanani @tedzhouhk

Reply:
I'm going to unwind all the python/lease changes. I don't understand them, and they aren't strictly necessary for what I'm trying to do.